test/unittest/rtps/persistence/PersistenceGuidTests.cpp

```diff
@@ -0,0 +1,435 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/domain/qos/DomainParticipantQos.hpp>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+
+#include <fastdds/rtps/participant/RTPSParticipant.h>
+#include <fastdds/rtps/writer/RTPSWriter.h>
+#include <fastdds/rtps/reader/RTPSReader.h>
+#include <fastdds/rtps/history/WriterHistory.h>
+#include <fastdds/rtps/RTPSDomain.h>
+#include <fastrtps/attributes/TopicAttributes.h>
+#include <fastrtps/xmlparser/XMLProfileManager.h>
+
+#include <climits>
+#include <tinyxml2.h>
+#include <gtest/gtest.h>
+
+using namespace eprosima::fastrtps::rtps;
+using namespace eprosima::fastdds::dds;
+
+class HelloWorld
+{
+public:
+
+    HelloWorld()
+    {
+    }
+
+    ~HelloWorld()
+    {
+    }
+
+    inline void message(
+            const std::string& _message)
+    {
+        m_message = _message;
+    }
+
+    inline const std::string& message() const
+    {
+        return m_message;
+    }
+
+    inline std::string& message()
+    {
+        return m_message;
+    }
+
+private:
+    std::string m_message;
+};
+
+class TopicDataTypeMock : public TopicDataType
+{
+public:
+
+    typedef HelloWorld type;
+
+    TopicDataTypeMock()
+        : TopicDataType()
+    {
+        setName("persistence_topic_type");
+    }
+
+    bool serialize(
+            void* /*data*/,
+            SerializedPayload_t* /*payload*/) override
+    {
+        return true;
+    }
+
+    bool deserialize(
+            SerializedPayload_t* /*payload*/,
+            void* /*data*/) override
+    {
+        return true;
+    }
+
+    std::function<uint32_t()> getSerializedSizeProvider(
+            void* /*data*/) override
+    {
+        return std::function<uint32_t()>();
+    }
+
+    void* createData() override
+    {
+        return nullptr;
+    }
+
+    void deleteData(
+            void* /*data*/) override
+    {
+    }
+
+    bool getKey(
+            void* /*data*/,
+            InstanceHandle_t* /*ihandle*/,
+            bool /*force_md5*/) override
+    {
+        return true;
+    }
+
+};
+
+
+/*!
+ * @fn TEST(PersistenceGuidTest, DDS_persistence_guid)
```

Comment:
In general, I think these descriptions could provide more details. Mainly "the test does this, then that, and then checks this way. It fails if this or that happens"